src/SharedUtilities/models.jl

```diff
@@ -100,21 +137,49 @@ function make_update_aux(model::AbstractModel)
 end
 
 """
-    make_ode_function(model::AbstractModel)
+    make_imp_tendency(model::AbstractImExModel)
 
-Returns an `ode_function` that updates auxiliary variables and
-updates the prognostic state.
+Returns an `imp_tendency` that updates auxiliary variables and
+updates the prognostic state of variables that are stepped implicitly.
 
-`ode_function!` should be compatible with OrdinaryDiffEq.jl solvers.
+`compute_imp_tendency!` should be compatible with OrdinaryDiffEq.jl solvers.
 """
-function make_ode_function(model::AbstractModel)
-    rhs! = make_rhs(model)
+function make_imp_tendency(model::AbstractImExModel)
+    compute_imp_tendency! = make_compute_imp_tendency(model)
     update_aux! = make_update_aux(model)
-    function ode_function!(dY, Y, p, t)
+    function imp_tendency!(dY, Y, p, t)
         update_aux!(p, Y, t)
-        rhs!(dY, Y, p, t)
+        compute_imp_tendency!(dY, Y, p, t)
     end
-    return ode_function!
+    return imp_tendency!
+end
+
+"""
+    make_imp_tendency(model::AbstractModel)
+
+Returns an `imp_tendency` that does nothing. This model type is not
+stepped explicity.
+"""
+function make_imp_tendency(model::AbstractModel)
+    function imp_tendency!(dY, Y, p, t) end
```

Comment:
Should this be `nothing`, or a function that doesnt do anything?

Reply:
Do you mean defining `make_imp_tendency` without the `function imp_tendency!` inside it at all? Or are you asking about `imp_tendency!`?

Reply:
After discussing offline, Kat and I decided to have `make_imp_tendency` return a function which does nothing. The advantage of this is that all methods of `make_imp_tendency` return a `Function` type, rather than `Union{Function, Nothing}`. A disadvantage of this is an unnecessary function call when `imp_tendency!` is called by the solver, but we figured any performance hit due to this will be minor.